ext/pcntl/pcntl.c

```diff
@@ -617,6 +622,54 @@ PHP_FUNCTION(pcntl_wstopsig)
 }
 /* }}} */
 
+#ifdef HAVE_EXECVEAT
+static zend_always_inline zend_result php_execve(const char *path, char **argv, char **envp) {
+	int fd = open(path, O_RDONLY | O_CLOEXEC);
+	if (fd < 0) {
+		return FAILURE;
+	}
+#ifdef AT_EXECVE_CHECK
+	// Linux >= 6.14 allows to check if `path` is allowed 
+	// for execution per kernel security policy (w/o actually running it)
+	if (execveat(fd, "", argv, envp, AT_EMPTY_PATH | AT_EXECVE_CHECK) < 0) {
```

Comment:
I'm not sure why we'd use `execveat` here. IIRC it's mainly for being able to look at a binary's contents before executing to avoid a TOCTOU issue and for generating executables that live entirely in a non-FS backed FD. If we don't do these, why not just use `execv(e)`?

Reply:
not sure what you mean ? just removing the above check ?

Reply:
I mean, why `execveat` versus `execve` if we aren't doing anything like those. (I probably should have put this comment over the signature instead of the body instead.)

Reply:
OK, after reading up I'm assuming we're doing this just for [`AT_EXECVE_CHECK`](https://lwn.net/Articles/1005958/). I'm assuming the only thing it's doing is moving log messages from saying whatever interpreter couldn't execute something to PHP failed to execute something. I guess we could also be doing this if we're being invoked as an interpreted if I read that first justification correctly as well.